revpi-hat-eep/src/bin/revpi-eep.rs

```diff
@@ -209,12 +207,66 @@ fn main() {
         }
     };
 
-    let edate = match cli.edate {
-        Some(edate) => edate,
-        None => chrono::Local::today().naive_local(),
+    let serial = if let Some(serial_cli) = cli.serial {
+        if let Some(serial_config) = config.serial {
+            eprintln!(
+                "WARNING: Overriding serial from the config file (`{}`) \
+                with the serial from the program arguments (`{}`).",
+                serial_config,
+                serial_cli
+            );
+        }
+        serial_cli
+    } else if let Some(serial_config) = config.serial {
+        serial_config
+    } else {
+        eprintln!("ERROR: The `serial` was not specified as argument or in the config file.");
+        process::exit(1);
+    };
+
+    let edate = if let Some(edate_cli) = cli.edate {
+        if let Some(edate_config) = config.edate {
+            eprintln!(
+                "WARNING: Overriding edate from the config file (`{}`) \
+                with the edate from the program arguments (`{}`).",
+                edate_config,
+                edate_cli
+            )
+        }
+        edate_cli
+    } else if let Some(edate_config) = config.edate {
+        edate_config
+    } else {
+        chrono::Local::today().naive_local()
     };
 
-    let eep = match create_rpi_eep(config, cli.serial, edate, cli.mac) {
+    let mac = if let Some(mac_cli) =  cli.mac {
+        if let Some(mac_config) = config.mac {
+            eprintln!(
+                "WARNING: Overriding mac from the config file (`{}`) \
+                with the mac from the program arguments (`{}`).",
+                mac_config.to_hex_string(),
+                mac_cli.to_hex_string()
+            );
+        }
+        mac_cli
+    } else if let Some(mac_config) = config.mac {
+        mac_config
+    } else {
+        eprintln!("ERROR: The `mac` was not specified as argument or in the config file.");
+        process::exit(1);
+    };
+
+    // Create a full config
```

Comment:
Existing values in config will be overwritten? Would prefer a clear comment, if so.

Reply:
This is an intermediate commit. The application behaves the same. The only thing which changes is that you can get a full config "backup/export". So no need to be more verbose here.

Reply:
Then why not remove the comment completely?